### ISSUE TITLE
nosqlunit creates invalid MongoClient connection when using Spring

### ISSUE THREAD

**Issue by gclayburg** [lordofthejars/nosql-unit#164]
Spring Boot 1.4.1

I am using this junit rule:
```
@Rule
public MongoDbRule mongoDbRule = newMongoDbRule().defaultSpringMongoDb("demo-test");

```
My test methods are working correctly.  Nosqlunit is using the Mongo bean defined in the spring context.  I have this defined elsewhere using a Fongo in-memory database.  My application code uses this same Mongo Bean and the defined "demo-test" database.   However, I am now getting many misleading error messages during a build:

```
2016-10-19 11:33:40,137 [}-localhost:27017] INFO  org.mongodb.driver.cluster           - Exception in monitor thread while connecting to server localhost:27017
com.mongodb.MongoSocketOpenException: Exception opening socket
	at com.mongodb.connection.SocketStream.open(SocketStream.java:63)
...
```

I traced these down to `MongoDbConfigurationBuilder.build()`.  This method tries to create a new MongoClient during the Junit rule, even though later on `SpringMongoDbRule` correctly grabs the already defined Mongo bean from the spring application context.

**Comment by choda** [comment]
A solution to get rid of the exception was given in thread #144.
This is only a workaround though (aka putting dust under the rug), for it creates a extra MongoClient that isn't actually used.

**Comment by gclayburg** [comment]
@choda Yup, this is exactly what I am seeing.  The workaround posted there works for me.  As you probably have noticed, there are several related issues here when configuring nosql-unit under spring boot.  Some tools require MongoClient from the Mongo 3.0 driver, and others like nosqlunit require a Mongo instance in the spring application context.   

I may just create a simple project to show how I combined what you and others have suggested to simplify this configuration so all these parts play better together.

**Comment by gclayburg** [comment]
Ok, I created a github project to show all the workarounds I use to get nosql-unit, Spring Data MongoDB, Fongo in-memory database and Javers working together.

https://github.com/gclayburg/nosqlunit-javers-demo

The more I loook at this the more I think it would be nice to have a spring boot starter module for both Fongo and nosql-unit.  It could make this integration work a little smoother.  Less boilerplate config is a good thing.
